### PR TITLE
Fix in case 'type' allows more than two types

### PR DESCRIPTION
**Pull request by funkyfuture**
there was an error in my implementation, leading to false negatives.

on that occasion:
- small code-cleanups
- output the errors in case an `assertSuccess` fails

### CODE REVIEW

File: cerberus/tests/tests.py [hunk 1]
```diff
@@ -831,3 +831,37 @@ def test_environment(self):
...
+
+        document = {'image': 'busybox',
+                    'extra_hosts': 'www.domain.net:127.0.0.1'}
+        self.assertSuccess(document, schema)
+
+        document = {'image': 'busybox',
+                    'extra_hosts': ['www.domain.net:127.0.0.1']}
+        self.assertSuccess(document, schema)
+
+        document = {'image': 'busybox',
```
Comment (nicolaiarocci) [inline cerberus/tests/tests.py]: Isn't this redundant (see above test)?

**Comment by funkyfuture** [comment]
thanks for the spot, it was even missing a test-case.

**Comment by funkyfuture** [comment]
pushed the fix.

**Comment by nicolaiarocci** [comment]
thanks!

**Pull request merged**